docs/topics/class-based-views/advanced.txt

```diff
@@ -0,0 +1,694 @@
+
+The ``get()`` method is a good example of flattening and simplifying the method
+hierarchy: we only need to get the forms into the context, so we can ignore
+``get_context_data()`` and all form-related methods and render the dictionary
+of forms directly to response.
+
+In ``post()`` we're getting the name of the form, instantiating it, and passing
+both the name and form to ``form_valid()`` or ``form_invalid()`` methods.
+
+In ``form_valid()`` we may wish to do custom processing based on the form name.
+Note that we've also flattened the method hierarchy by redirecting directly the
+the success url.
+
+The ``form_invalid()`` method requires a bit of explanation: it's very similar
+to the ``get()`` method but we need to create fresh instances of all forms
+except for the submitted one, which is inserted into context as a bound form.
+
+.. snippet::
+    :filename: views.py
+
+    class CommentForm(forms.Form):
+        comment = forms.CharField()
+
+    class NFormsView(FormView):
+        form_classes = dict(
+                            comment_form = CommentForm,
+                            author_form = AuthorInterestForm,
+                            )
+        template_name = "two-forms.html"
+
+        def get(self, request, *args, **kwargs):
+            context = {n: cls() for n, cls in self.form_classes.items()}
+            return self.render_to_response(context)
+
+        def post(self, request, *args, **kwargs):
+            for name, cls in self.form_classes.items():
+                if name in request.POST:
+                    form = self.get_form(cls)
+                    break
+
+            if form.is_valid():
+                return self.form_valid(name, form)
+            else:
+                return self.form_invalid(name, form)
+
+        def form_valid(self, name, form):
+            # Here, we would have some logic based on user's form inputs
+            return redirect('two-forms')
+
+        def form_invalid(self, name, form):
+            context = {n: cls() for n, cls in self.form_classes.items()}
+            context.update({name: form})
+            return self.render_to_response(context)
+
+The key change in the template is that each form's ``input`` tag needs to
+specify the form's name:
+
+.. code-block:: html+django
+
+    <h2>Author form</h2>
+
+    <form action="" method="post" accept-charset="utf-8">{% csrf_token %}
+        <table border="0" cellpadding="2" cellspacing="2">
+            {{ author_form.as_table }}
+        </table>
+    <p><input type="submit" name="author_form" value="Submit" /></p>
+    </form>
+
+    <h2>Comment form</h2>
+
+    <form action="" method="post" accept-charset="utf-8">{% csrf_token %}
+        <table border="0" cellpadding="2" cellspacing="2">
+            {{ comment_form.as_table }}
+        </table>
+    <p><input type="submit" name="comment_form" value="Submit" /></p>
+    </form>
+
+Create / Update View
+====================
```

Comment:
I don't see the point of this view really. What URL design is going to want the exact same view here? Or are you pointing both URLs at the same view? In that case do your customisations in your own mixin and then add it to create and update base views separately.